pkg/controllers/managedzone/managedzone_controller.go

```diff
@@ -182,7 +185,11 @@ func (r *ManagedZoneReconciler) deleteManagedZone(ctx context.Context, managedZo
 		return nil
 	}
 
-	err := r.DNSProvider.DeleteManagedZone(managedZone)
+	DNSProvider, err := r.DNSProvider(ctx, managedZone)
+	if err != nil {
```

Comment:
same here, adding a status condition on the zone if we fail to load the provider. Otherwise we will error out in the logs etc but the user will likely not have an external info letting them know what is happening